app/graphql/resolvers/base_resolver.rb

```diff
@@ -42,6 +42,10 @@ def matching_user(submission, session_id)
       submitted_by_current_session?(submission, session_id)
   end
 
+  def matching_email(submission, user_email)
+    submission.user_email == user_email
```

Comment:
suggestion: It may be good to `downcase` the email addresses to make sure they match.